pkg/plan/build_releases.go

```diff
@@ -45,15 +47,33 @@ func addToPlan(
 		}
 	}
 
+	// Run hooks
+	lifeCycle := rel.LifeCycle()
+	err = lifeCycle.RunPreBuild(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		lifecycleErr := lifeCycle.RunPostBuild(ctx)
+		if lifecycleErr != nil {
+			log.Errorf("got an error from postbuild hooks: %v", lifecycleErr)
+			if err == nil {
+				err = lifecycleErr
+			}
+		}
+	}()
```

Comment:
I am not sure this is the place both of these lifecycles should be run.

IMO:

- per-release pre-build should be run after `p.buildReleases(o.Tags, o.MatchAll)` (so as soon as we know list of releases, we run the hook)
- per-release post-build should be run in the end of `*Plan.Build` (after everything was built)

Reply:
`addToPlan` happens during `p.buildReleases`, and will make pre-build hook to be executed in _dependency order_. 

But after you mentioned I found that releases in `plan` are already in dependency order. So indeed it can be run after `p.buildReleases`.

Reply:
@r3nic1e  I've altered the hook point according to your comment.